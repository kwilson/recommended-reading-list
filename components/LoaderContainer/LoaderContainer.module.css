.container {
  display: flex;
  flex-direction: column;
}

.isLoading {
  align-items: center;
  background: #efefef;
  justify-content: center;
  min-height: 10rem;
}
