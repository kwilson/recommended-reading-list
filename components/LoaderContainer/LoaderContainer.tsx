import { FunctionComponent } from 'react';

interface IProps {
  loading: boolean;
}

import styles from './LoaderContainer.module.css';

export const LoaderContainer: FunctionComponent<IProps> = ({
  loading,
  children,
}) => {
  const classNames = [styles.container, loading && styles.isLoading]
    .filter(Boolean)
    .join(' ');

  return (
    <div className={classNames}>
      {loading && (
        <div className="notification" data-testid="loader">
          Loading...
        </div>
      )}
      {!loading && children}
    </div>
  );
};
