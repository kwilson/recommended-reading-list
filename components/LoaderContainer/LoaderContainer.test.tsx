import { render, screen } from '@testing-library/react';

import { LoaderContainer } from './LoaderContainer';

describe(`<${LoaderContainer.name} />`, () => {
  const ChildItem = () => <p data-testid="child-item-content">Child Item</p>;

  it('renders as expected when loading', async () => {
    const { asFragment } = render(
      <LoaderContainer loading>
        <ChildItem />
      </LoaderContainer>
    );

    expect(screen.getByTestId('loader')).toBeDefined();
    expect(screen.queryByTestId('child-item-content')).toBe(null);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as expected when not loading', async () => {
    const { asFragment } = render(
      <LoaderContainer loading={false}>
        <ChildItem />
      </LoaderContainer>
    );

    expect(screen.getByTestId('child-item-content')).toBeDefined();
    expect(screen.queryByTestId('loader')).toBe(null);

    expect(asFragment()).toMatchSnapshot();
  });
});
