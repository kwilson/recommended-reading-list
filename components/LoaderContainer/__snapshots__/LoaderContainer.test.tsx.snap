// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderContainer /> renders as expected when loading 1`] = `
<DocumentFragment>
  <div
    class="container isLoading"
  >
    <div
      class="notification"
      data-testid="loader"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;

exports[`<LoaderContainer /> renders as expected when not loading 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p
      data-testid="child-item-content"
    >
      Child Item
    </p>
  </div>
</DocumentFragment>
`;
